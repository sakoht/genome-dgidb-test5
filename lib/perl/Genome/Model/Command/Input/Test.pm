
###########################################################################

package Genome::Model::Command::Input::Test;

use strict;
use warnings;

use base 'Test::Class';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input';
}



###########################################################################

package Genome::Model::Command::Input::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;
require Genome::Model::Test;
require File::Temp;
require File::Path;

sub valid_param_sets {
    return grep { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub invalid_param_sets {
    return map { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_mock_model(
            type_name => 'tester',
            instrument_data_count => 0,
        )
            or confess "Can't create mock tester model.";
    }

    return $self->{_model};
}

sub _build {
    my $self = shift;

    return ($self->_model->builds)[0];
}



sub startup : Tests(startup => no_plan) {
    my $self = shift;
    
    no warnings;
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    use warnings;
    is_deeply(
        [ Genome::Model::Command->get_model_type_names ],
        [qw/ tester /],
        'Set tester type name.',
    );

    $self->_startup or confess "Failed startup.";
    
    return 1;
}

sub _startup {
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Add::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Add';
}

sub _valid_param_sets {
    return (
        { # add multiple 'value_ids' 
            name => 'friends',
            ids => 'Watson,Crick',
            before_execute => '_before_execute',
            after_execute => '_after_execute',
        },
        { # add single 'value'
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
            before_execute => '_before_execute',
            after_execute => '_after_execute',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to add a not 'is_many' input
            name => 'coolness', 
            ids => 'none',
        },
        { # try to add Crick again
            name => 'friends',
            ids => 'Crick',
        },
        { # try to add id that don't exist
            name => 'inst_data',
            ids => '2sep09.934noexist',
        },
    );
}

sub _before_execute {
    my ($self, $obj, $param_set) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    ok(!@ids, "No $name found for model.");
    
    return 1;
}

sub _after_execute {
    my ($self, $obj, $param_set) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    ok(@ids, "Added $name to model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Remove::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Genome::Utility::TestBase;
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Remove';
}

sub _valid_param_sets {
    return (
        { # remove multiple 'value_ids' 
            name => 'friends',
            ids => 'Watson,Crick',
        },
        { # remove single 'value'
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
            abandon_builds => 1,
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to remove a not 'is_many' input
            name => 'coolness', 
            ids => 'none',
        },
        { # try to remove input that is not linked to the model anymore
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
        },
    );
}

sub friends {
    return 'Crick';
}

sub inst_data {
    return Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
}

sub _startup {
    my $self = shift;

    my $model = $self->_model;

    $model->add_friend('Crick');
    $model->add_friend('Watson');
    my @friends = $model->friends();
    is_deeply(\@friends, [qw/ Crick Watson /], 'Added friends to remove.');
    my $id = Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
        or confess "Can't get sange id 2sep09.934pmaa1";
    $model->add_inst_data($id);
    my @inst_data = $model->inst_data;
    is_deeply(\@inst_data, [ $id ], 'Added instr_data to remove');
    
    my $build = $self->_build;
    my $build_input = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::Model::Build::Input',
        name => 'instrument_data',
        value_id => '2sep09.934pmaa1',
        build_id => $build->id,
    );
    $build->mock('abandon', sub{ return $build->build_event->event_status('Abandoned'); });

    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    #print Dumper(\@ids);
    ok(!@ids, "Removed $name from model.");
    if ( $obj->abandon_builds ) {
        is($self->_build->status, 'Abandoned', 'Successfully abandoned build');
    }

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::List::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::List';
}

sub valid_param_sets {
    return (
        {
            filter => 'model_id='.$_[0]->_model->id,
        },
    );
}

sub invalid_param_sets {
}

sub _startup {
    my $self = shift;

    $self->_model->coolness('low');
    is($self->_model->coolness, 'low', 'Set coolness (input)');

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Names::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Names';
}

sub valid_param_sets {
    return (
        {
            type_name => 'tester',
        },
    );
}

sub invalid_param_sets {
    return (
        {
            type_name => 'none',
        },
    );
}

sub startup : Tests(startup) {
    my $self = shift;
    
    no warnings 'once';
    no warnings 'redefine';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Update::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Update';
}

sub _valid_param_sets {
    return (
        { # remove single 'value'
            name => 'coolness',
            value => 'moderate',
        },
        { # remove single 'value'
            name => 'coolness',
            value => 'UNDEF',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to update an 'is_many' input
            name => 'friends',
            value => 'Watson',
        },
    );
}

###########################################################################

1;

=pod

=head1 Tests

=head1 Disclaimer

 Copyright (C) 2009 Washington University Genome Sequencing Center

 This script is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
 License for more details.

=head1 Author(s)

 Eddie Belter <ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

