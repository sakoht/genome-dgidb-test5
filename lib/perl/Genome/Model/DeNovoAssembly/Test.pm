package Genome::Model::DeNovoAssembly::Test;

use strict;
use warnings;

require Carp;
use Data::Dumper 'Dumper';
require File::Temp;
require Genome::Model::Test;
require Genome::Utility::TestBase;

BEGIN {
    $ENV{UR_USE_DUMMY_AUTOGENERATED_IDS} = 1;
    $ENV{UR_DBI_NO_COMMIT} = 1;
}

#< Dirs >#
sub _base_directory {
    return '/gsc/var/cache/testsuite/data/Genome-Model/DeNovoAssembly';
}

sub _example_directory_for_model {
    my ($self, $model) = @_;

    Carp::confess "No model to get example directory" unless $model;
    
    my %dirs_versions = (
        soap_solexa => '9',
        velvet_solexa => '0.3',
        abyss_solexa => '1'
    );

    my $assembler_platform = $model->processing_profile->assembler_base_name.'_'.$model->sequencing_platform;
    my $dir = $self->_base_directory.'/'.$assembler_platform.'_build_v'.$dirs_versions{$assembler_platform};

    Carp::confess("Example directory ($dir) for de novo assembly model does not exist.") unless -d $dir;
    
    return $dir;
}

sub _instrument_data_dir {
    return $_[0]->_base_directory.'/inst_data/';
}
#<>#

#< Library Taxon >#
our($taxon, $sample, $library);
sub _create_taxon_sample_and_library {
    my $class = shift;

    $taxon = Genome::Taxon->create(
        name => 'Escherichia coli TEST',
        domain => 'Bacteria',
        current_default_org_prefix => undef,
        estimated_genome_size => 4500000,
        current_genome_refseq_id => undef,
        ncbi_taxon_id => undef,
        ncbi_taxon_species_name => undef,
        species_latin_name => 'Escherichia coli',
        strain_name => 'TEST',
    );
    if ( not $taxon ) {
        Carp::confess('Cannot create taxon');
    }

    $sample = Genome::Sample->create(
        id => -1234,
        name => 'TEST-000',
        taxon_id => $taxon->id,
    );
    if ( not $sample ) {
        Carp::confess('Cannot create sample');
    }

    $library = Genome::Library->create(
        id => -12345,
        name => $sample->name.'-testlibs',
        sample_id => $sample->id,
    );
    if ( not $library ) { 
        Carp::confess('Cannot create library');
    }

    return 1;
}

sub taxon {
    my $class = shift;
    if ( not $taxon ) {
        $class->_create_taxon_sample_and_library;
    }
    return $taxon;
}

sub library {
    my $class = shift;
    if ( not $library ) {
        $class->_create_taxon_sample_and_library;
    }
    return $library;
}
#<>#

#< PP Models and Builds >#
sub processing_profile_for_soap {
    my $class = shift;

    my $pp = Genome::ProcessingProfile::DeNovoAssembly->create(
        name => 'De Novo Assembly Soap Test',
        sequencing_platform => 'solexa',
        assembler_name => 'soap de-novo-assemble',
        assembler_version => '1.04',
        assembler_params => '-kmer_size 31 -resolve_repeats -kmer_frequency_cutoff 1',
        read_processor => 'trimmer bwa-style -trim-qual-level 10 | filter by-length --filter-length 35 | rename illumina-to-pcap',
        post_assemble => 'standard-outputs',
    );

    if ( not $pp ) {
        Carp::confess('Cannot create de novo assembly processing profile for soap');
    }

    return $pp;
}

sub processing_profile_for_velvet {
    my $class = shift;

    my $pp = Genome::ProcessingProfile::DeNovoAssembly->create(
        name => 'De Novo Assembly Velvet Test',
        sequencing_platform => 'solexa',
        coverage => 0.5,#25000,
        assembler_name => 'velvet one-button',
        assembler_version => '0.7.57-64',
        assembler_params => '-hash_sizes 31 33 35 -min_contig_length 100',
        read_processor => 'trimmer by-length -trim-length 10 | rename illumina-to-pcap',
        post_assemble => 'standard-outputs',
    );

    if ( not $pp ) {
        Carp::confess('Cannot create de novo assembly processing profile for velvet');
    }

    return $pp;
}

sub processing_profile_for_abyss {
    my $class = shift;

    my $pp = Genome::ProcessingProfile::DeNovoAssembly->create(
        name => 'De Novo Assembly Abyss Test',
        sequencing_platform => 'solexa',
        assembler_name => 'abyss parallel',
        assembler_version => '1.2.7',
        assembler_params => '-kmer_size 25,31..35 step 2,50 -num_jobs 4',
    );

    if ( not $pp ) {
        Carp::confess('Cannot create de novo assembly processing profile for abyss');
    }

    return $pp;
}

sub _model_for_assembler {
    my ($class, $assembler) = @_;

    Carp::confess('No assembler given to get model') if not $assembler;

    my $taxon = $class->taxon or die; # confesses
    my $pp_method = 'processing_profile_for_'.$assembler;
    my $pp = $class->$pp_method or die; # confesses
    my $tmpdir = File::Temp::tempdir(CLEANUP => 1);
    my $model = Genome::Model::DeNovoAssembly->create(
        processing_profile => $pp,
        subject_name => $taxon->name,
        subject_type => 'species_name',
        center_name => 'WUGC',
        data_directory => $tmpdir,
    );
    if ( not $model ) {
        Carp::confess('Cannot create de novo assembly model for '.$assembler);
    }

    my $instrument_data = $class->solexa_instrument_data;
    if ( not $instrument_data ) {
        Carp::confess('Cannot get solexa instrument data');
    }

    my $add_ok = $model->add_instrument_data($instrument_data);
    if ( not $add_ok ) {
        Carp::confess('Cannot add instrument data to model');
    }

    my @instrument_data = $model->instrument_data;
    if ( not @instrument_data ) {
        Carp::confess('Cannot get instrument data from model');
    }

    return $model;
}

sub model_for_soap {
    return $_[0]->_model_for_assembler('soap');
}

sub model_for_velvet {
    return $_[0]->_model_for_assembler('velvet');
}

sub model_for_abyss {
    return $_[0]->_model_for_assembler('abyss');
}

sub example_build_for_model {
    my ($class, $model) = @_;

    Carp::confess('No de novo assembly model to create example build') if not $model;

    my $dir = $class->_example_directory_for_model($model) or die;
    my $build = Genome::Model::Build->create(
        model=> $model,
        data_directory => $dir,
    );
    if ( not $build ) {
        Carp::confess('Cannot create de novo assembly build');
    }

    $build->status('Succeeded');
    $build->the_master_event->date_completed(UR::Time->now);

    return $build;
}
#<>#

#< Instruement Data >#
our $solexa_instrument_data;
sub solexa_instrument_data {
    my $class = shift;

    return $solexa_instrument_data if $solexa_instrument_data;

    my $library = $class->library or die; # confesses

    my $archive_path = $class->_instrument_data_dir.'/2854709902/sequence_61H93_8_CGATGT.tar.gz';
    if ( not -s $archive_path ) {
        Carp::confess("No archive path in example directory ($archive_path)");
    }

    $solexa_instrument_data = Genome::InstrumentData::Imported->create(
        id => -7777,
        sequencing_platform => 'solexa',
        import_format => 'fastq',
        import_source_name => 'WUGC',
        read_count => 30000,
        read_length => 100,
        subset_name => '8-CGATGT',
        is_paired_end => 1,
        library => $library,
        median_insert_size => 260,# 181, but 260 was used to generate assembly
        sra_sample_id => 'SRS000000',
    );

    if ( not $solexa_instrument_data ) {
        Carp::confess('Cannot create solexa instrument data');
    }

    my $alloc = Genome::Disk::Allocation->__define__(
        owner_id => $solexa_instrument_data->id,
        owner_class_name => $solexa_instrument_data->class,
        disk_group_name => 'info_alignments',
        mount_path => '/gsc/var/cache/testsuite/data/Genome-Model/DeNovoAssembly',
        group_subdirectory => 'inst_data',
        allocation_path => $solexa_instrument_data->id,
        absolute_path => '/gsc/var/cache/testsuite/data/Genome-Model/DeNovoAssembly/inst_data/2854709902',
    );
    if (not $alloc ) {
        Carp::confess('Could not create disk allocation for instrument data');
    }

    if ( not -s $solexa_instrument_data->archive_path ) {
        Carp::confess('Cannot get archive path from instrument data');
    }

    return $solexa_instrument_data;
}
#<>#

1;

