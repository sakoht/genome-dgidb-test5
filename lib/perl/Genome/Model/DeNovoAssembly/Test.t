#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Data::Dumper 'Dumper';
use Test::More;

use_ok('Genome::Model::DeNovoAssembly::Test') or die;

my $rv;

#< PP >#
# fail - no seq platform
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model();
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - get w/o seq platform',
);
# fail - no assembler
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/o assembler',
);
# fail - invalid seq platform/assembler combo
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        assembler_name => 'consed',
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/ invalid seq paltform/assembler combo',
);

# valid velvet solexa
my $velvet_model = Genome::Model::DeNovoAssembly::Test->get_mock_model(
    assembler_name => 'velvet',
    sequencing_platform => 'solexa',
);
ok($velvet_model, 'Got mock DNA model for velvet solexa') or die;
my @inst_data = $velvet_model->instrument_data;
ok(@inst_data, 'Model instrument data');
ok(-s $inst_data[0]->archive_path, 'Solexa archive path exists');

# valid newbler 454
my $newbler_model = Genome::Model::DeNovoAssembly::Test->get_mock_model(
    assembler_name => 'newbler',
    sequencing_platform => 454,
);
ok($newbler_model, 'Got mock DNA model for newbler 454') or die;

done_testing();
exit;

#$HeadURL$
#$Id$
