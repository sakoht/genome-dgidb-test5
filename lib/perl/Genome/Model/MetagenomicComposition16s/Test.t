#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Data::Dumper 'Dumper';
use Test::More;

use_ok('Genome::Model::MetagenomicComposition16s::Test') or die;

my $rv;

#< PP >#
# fail - no seq platform
eval{
    $rv = Genome::Model::MetagenomicComposition16s::Test->get_mock_model();
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - get w/o seq platform',
);
# fail - invalid seq platform
eval{
    $rv = Genome::Model::MetagenomicComposition16s::Test->get_mock_model(
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/ invalid sequencing platform',
);

# valid 
my $model = Genome::Model::MetagenomicComposition16s::Test->get_mock_model(
    sequencing_platform => '454',
);
ok($model, 'Got mock DNA model') or die;
my @inst_data = $model->instrument_data;
ok(@inst_data, 'Model instrument data');

done_testing();
exit;

#$HeadURL$
#$Id$
