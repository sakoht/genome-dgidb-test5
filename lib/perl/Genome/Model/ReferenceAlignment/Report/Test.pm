package Genome::Model::ReferenceAlignment::Report::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

use Data::Dumper 'Dumper';
require Genome::Model::Test;
use Test::More;

sub generator {
    return $_[0]->{_object};
}

sub base_params_for_test_class {
    return (
        build_id => $_[0]->_mock_model->last_complete_build->id,
    );
}

sub required_params_for_class {
    return (qw/ build_id /);
}

sub _mock_model {
    my $self = shift;

    unless ( $self->{_mock_model} ) { 
        $self->{_mock_model} = Genome::Model::Test->create_mock_model(
            type_name => 'reference alignment solexa',
        )
            or die;
    }

    return $self->{_mock_model};
}

sub skip_generate { return 0; }

#< Test Methods >#
sub test01_generate_report : Tests {
    my $self = shift;

    if ( $self->skip_generate ) {
        # Skipping generate...too many methods and data to mock.  This was not tested before
        can_ok($self->test_class, '_add_to_report_xml');
    }
    else {
        my $report = $self->generator->generate_report;
        ok($report, 'Generated report');
        #print Dumper($report);
    }
    
    # Save...maybe?  For comparison?
    # $report->name('Saved '.$report->name);
    # ok($self->report->save, 'Saved report');

    return 1;
}

#######################################################################

package Genome::Model::ReferenceAlignment::Report::DbSnpTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::DbSnp';
}

sub params_for_test_class {
    return (
        name => 'Db Snp',
        #override_model_snp_file => '/gsc/var/cache/testsuite/data/Genome-Model-ReferenceAlignment/build-50001/maq_snp_related_metrics/all.snps',
        $_[0]->base_params_for_test_class,
    );
}

#######################################################################

package Genome::Model::ReferenceAlignment::Report::GoldSnpTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::GoldSnp';
}

sub params_for_test_class {
    return (
        name => 'Gold Snp',
        $_[0]->base_params_for_test_class,
    );
}

sub skip_generate { return 1; }

#######################################################################

package Genome::Model::ReferenceAlignment::Report::SolexaStageOneTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub generator {
    return $_[0]->{_object};
}

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::SolexaStageOne';
}

sub params_for_test_class {
    return (
        $_[0]->base_params_for_test_class,
    );
}

sub skip_generate { return 1; }

#######################################################################

package Genome::Model::ReferenceAlignment::Report::SolexaStageTwoTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::SolexaStageTwo';
}

sub params_for_test_class {
    return (
        $_[0]->base_params_for_test_class,
    );
}

sub skip_generate { return 1; }

#######################################################################

package Genome::Model::ReferenceAlignment::Report::MapcheckTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::Mapcheck';
}

sub params_for_test_class {
    return (
        $_[0]->base_params_for_test_class,
    );
}

sub skip_generate { return 1; }

#######################################################################

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2008 Washington University Genome Sequencing Center

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

