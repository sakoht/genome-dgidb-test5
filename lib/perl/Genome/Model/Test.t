#! /gsc/bin/perl

# THIS TESTS Genome::Model::Test and MODEL MOCKING

use strict;
use warnings;

use above 'Genome';
use Carp::Always;
use Data::Dumper 'Dumper';
use Test::More 'no_plan';

use_ok('Genome::Model::Test');

# Model
my $model = Genome::Model::Test->create_mock_model(
    type_name => 'tester',
    instrument_data_count => 2,
);
ok($model, 'Created mock model');
is($model->id, $model->genome_model_id, 'model_id and genome_model_id match');

# PP 
ok($model->processing_profile, 'Model has a Processing profile');
is($model->processing_profile_id, $model->processing_profile->id, 'processing profile id');
is($model->type_name, $model->processing_profile->type_name, 'type name');
is($model->sequencing_platform, $model->processing_profile->sequencing_platform, 'sequencing_platform');
is($model->dna_source, $model->processing_profile->dna_source, 'dna_source');

# Subject
ok($model->processing_profile, 'Model has a subject');
is($model->subject_name, $model->subject->name, 'subject name');

# Builds
my @builds = $model->builds;
is_deeply(
    \@builds,
    [Genome::Model::Build->get(model_id => $model->id)],
    'Model\'s builds match Genome::Model::Build->get',
);
my $last_complete_build = $model->last_complete_build;
is_deeply($last_complete_build, $builds[0], 'last_complete_build');
is($last_complete_build->status, 'Succeeded', 'Build is succeeded');
is_deeply($model->last_succeeded_build, $last_complete_build, 'Last succeeded build');

# Ints Data
my @instrument_data_assignments = $model->instrument_data_assignments;
ok(@instrument_data_assignments, 'instrument_data_assignments');
my @instrument_data = $model->instrument_data;
is_deeply(
    \@instrument_data,
    [map { $_->instrument_data } $model->instrument_data_assignments],
    'instrument_data',
);

exit;

=pod

=head1 Disclaimer

Copyright (C) 2009 Washington University Genome Sequencing Center

This script is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> <ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$
