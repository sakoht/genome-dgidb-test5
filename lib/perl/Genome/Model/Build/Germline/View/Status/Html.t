#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Test::More tests => 6;

use_ok('Genome::Model::Build::Germline::View::Status::Html') or die "test cannot continue...";

#2742999043 is M_BQ-081218_rna_mouseAPL_tube1

#102002447

my $subject = Genome::Model::Build->get(102002447);
ok($subject, "found expected build subject") or die "test cannot continue...";

my $view_obj = $subject->create_view(
    xsl_root => Genome->base_dir . '/xsl',
    rest_variable => '/cgi-bin/rest.cgi',
    toolkit => 'html',
    perspective => 'status',
); 
ok($view_obj, "created a view") or die "test cannot continue...";
isa_ok($view_obj, 'Genome::Model::Build::Germline::View::Status::Html');

my $html = $view_obj->_generate_content();
ok($html, "view returns HTML") or die "test cannot continue...";

SKIP: {
    skip "No Html.t.expected in place.",1;
    my @diff =
        grep { $_ !~ /generated-at/ }
        grep { /\w/ }
        Genome::Sys->diff_file_vs_text(__FILE__ . '.expected',$html);
    
    is("@diff","","HTML has no differences from expected value");
}
