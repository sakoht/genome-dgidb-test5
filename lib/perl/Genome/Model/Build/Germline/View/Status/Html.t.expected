<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Status </title>
<link rel="shortcut icon" href="/resources/report_resources/apipe_dashboard/images/gc_favicon.png" type="image/png">
<link rel="stylesheet" href="/resources/report_resources/apipe_dashboard/css/master.css" type="text/css" media="screen">
<link rel="stylesheet" href="/resources/report_resources/apipe_dashboard/css/tablesorter.css" type="text/css" media="screen">
<script type="text/javascript" src="/resources/report_resources/jquery/jquery.js"></script><script type="text/javascript" src="/resources/report_resources/jquery/jquery.tablesorter.min.js"></script>
</head>
<body><div class="container"><div class="background">
<div class="page_header"><table cellpadding="0" cellspacing="0" border="0"><tr>
<td><a href="status.cgi" alt="Go to Search Page" title="Go to Search Page"><img src="/resources/report_resources/apipe_dashboard/images/gc_header_logo2.png" width="44" height="45" align="absmiddle"></a></td>
<td><h1>Analysis Reports v0.2</h1></td>
</tr></table></div>
<div class="page_padding">
<script type="text/javascript">
          
              $(document).ready(function() {
                  $("#alignment").tablesorter({
                      // sort on first column, ascending
                      sortList: [[0,0]]
                  });
              });
          
        </script><script type="text/javascript" src="/resources/report_resources/jquery/boxy/src/javascripts/jquery.boxy.js"></script><link rel="stylesheet" href="/resources/report_resources/jquery/boxy/src/stylesheets/boxy.css" type="text/css">
<script type="text/javascript">
          
                   function event_popup(eventObject) {
                   // assemble event info into a table
                   var popup_content = '<table class="boxy_info" cellpadding="0" cellspacing="0" border="0"><tbody>';
                   for (prop in eventObject) {
                   if (prop != 'popup_title') {
                   popup_content += '<tr><td class="label">' + prop.replace(/_/g," ") + ':</td><td class="value">' + eventObject[prop] + '</td></tr>';
                   }
                   }

popup_content += '</tbody></table>';

// create popup
var popup = new Boxy(popup_content, {title:eventObject.popup_title, fixed:false});
popup.center();
}
          
        </script><h2 class="page_title build_icon">Build 102002447 Status</h2>
<table cellpadding="0" cellspacing="0" border="0" class="info_table_group"><tr>
<td><table border="0" cellpadding="0" cellspacing="0" class="info_table" width="100%">
<tr>
<td class="label">Model ID:</td>
<td class="value"><a xmlns:rest="urn:rest" href="/cgi-bin/rest.cgi/Genome/Model/Germline/status.html?id=2855070496">2855070496</a></td>
</tr>
<tr>
<td class="label">Model Name:</td>
<td class="value">tmooney_germline_test</td>
</tr>
<tr>
<td class="label">Base Alignment Path:</td>
<td class="value"><a href="https://gscweb.gsc.wustl.edu//gscmnt/sata920/info/model_data/2852955513/build101614583/alignments" class="grey">(base alignment path)</a></td>
</tr>
<tr>
<td class="label">LSF Job ID:</td>
<td class="value">1949492</td>
</tr>
<tr>
<td class="label">Workflow Instance ID:</td>
<td class="value"></td>
</tr>
</table></td>
<td><table border="0" cellpadding="0" cellspacing="0" class="info_table" width="100%">
<tr>
<td class="label">Status:</td>
<td class="value">Succeeded <a href="https://gscweb.gsc.wustl.edu//gscmnt/sata919/info/model_data/2855070496/build102002447/reports/Summary/report.html" class="grey">(view build summary report)</a>
</td>
</tr>
<tr>
<td class="label">Processing Profile:</td>
<td class="value">Metabolic_Disorders (germline)</td>
</tr>
<tr>
<td class="label">Data Directory:</td>
<td class="value"><a href="https://gscweb.gsc.wustl.edu//gscmnt/sata919/info/model_data/2855070496/build102002447">/gscmnt/sata919/info/model_data/2855070496/build102002447</a></td>
</tr>
<tr>
<td class="label">Disk Space (kbytes):</td>
<td class="value">30,144</td>
</tr>
<tr>
<td class="label">Error File:</td>
<td class="value"><a href="https://gscweb.gsc.wustl.edu//gscmnt/sata919/info/model_data/2855070496/build102002447/logs//102002478.err">102002478.err</a></td>
</tr>
</table></td>
</tr></table>
<table border="0" cellpadding="0" cellspacing="0" class="stages" width="100%"><tr></tr></table>
<hr>
</div>
</div></div></body>
</html>
